pkg/db/driver/bolt/store.go

```diff
@@ -0,0 +1,115 @@
+package bolt
+
+import (
+	bh "github.com/timshannon/bolthold"
+	bolt "go.etcd.io/bbolt"
+
+	"github.com/allez-chauffe/marcel/pkg/db/driver/driver"
+)
+
+type store struct {
+	*client
+	config *driver.StoreConfig
+	tx     *bolt.Tx
+}
+
+var _ driver.StoreBase = new(store)
+
+func (s *store) Get(id interface{}, e interface{}) (err error) {
+	if s.HasTransaction() {
+		err = s.bh.TxGet(s.tx, id, e)
+	} else {
+		err = s.bh.Get(id, e)
+	}
+
+	if err == bh.ErrNotFound {
+		err = driver.ErrNotFound
+	}
+
+	return
+}
+
+func (s *store) Insert(id interface{}, e interface{}) error {
+	if s.config.AutoIncrement() {
+		id = bh.NextSequence()
+	}
+
+	if s.HasTransaction() {
+		return s.bh.TxInsert(s.tx, id, e)
+	}
+	return s.bh.Insert(id, e)
+}
+
+func (s *store) Update(id interface{}, e interface{}) (err error) {
+	if s.HasTransaction() {
+		err = s.bh.TxUpdate(s.tx, id, e)
+	} else {
+		err = s.bh.Update(id, e)
+	}
+
+	if err == bh.ErrNotFound {
+		err = driver.ErrNotFound
+	}
+
+	return
+}
+
+func (s *store) Delete(id interface{}) error {
+	if s.HasTransaction() {
+		return s.bh.TxDelete(s.tx, id, s.config.New())
+	}
+	return s.bh.Delete(id, s.config.New())
+}
+
+func (s *store) WithTransaction(tx driver.Transaction) driver.StoreBase {
+	if tx == nil { // FIXME is this useful?
```

Comment:
IT's useful at least in tests, it allows to use the same test code for stores with and without transactions. Perhpas with the new driver architecture it's not needed any more and we shouldn't allow to use this function with a nil transaction